test/extensions/tradeExtension.spec.ts

```diff
@@ -0,0 +1,404 @@
+
+    // SetToken + DelegatedManager: TradeModule not initialized, TradeExtension not pending
+    setToken4 = await setV2Setup.createSetToken(
+      [setV2Setup.dai.address],
+      [ether(1)],
+      [setV2Setup.issuanceModule.address, tradeModule.address]
+    );
+
+    await setV2Setup.issuanceModule.initialize(setToken4.address, ADDRESS_ZERO);
+
+    delegatedManager4 = await deployer.manager.deployDelegatedManager(
+      setToken4.address,
+      factory.address,
+      methodologist.address,
+      [],
+      [operator.address],
+      [setV2Setup.usdc.address, setV2Setup.weth.address],
+      true
+    );
+
+    await setToken4.setManager(delegatedManager4.address);
+
+    // SetToken + DelegatedManager: No TradeModule, TradeExtension not pending
+    setToken5 = await setV2Setup.createSetToken(
+      [setV2Setup.dai.address],
+      [ether(1)],
+      [setV2Setup.issuanceModule.address]
+    );
+
+    await setV2Setup.issuanceModule.initialize(setToken5.address, ADDRESS_ZERO);
+
+    delegatedManager5 = await deployer.manager.deployDelegatedManager(
+      setToken5.address,
+      factory.address,
+      methodologist.address,
+      [],
+      [operator.address],
+      [setV2Setup.usdc.address, setV2Setup.weth.address],
+      true
+    );
+
+    await setToken5.setManager(delegatedManager5.address);
+
+    // SetToken + DelegatedManager: TradeModule initialized, TradeExtension initialized
+    setToken6 = await setV2Setup.createSetToken(
+      [setV2Setup.dai.address],
+      [ether(1)],
+      [setV2Setup.issuanceModule.address, tradeModule.address]
+    );
+
+    await setV2Setup.issuanceModule.initialize(setToken6.address, ADDRESS_ZERO);
+    await tradeModule.initialize(setToken6.address);
+
+    delegatedManager6 = await deployer.manager.deployDelegatedManager(
+      setToken6.address,
+      factory.address,
+      methodologist.address,
+      [tradeExtension.address],
+      [operator.address],
+      [setV2Setup.usdc.address, setV2Setup.weth.address],
+      true
+    );
+
+    await setToken6.setManager(delegatedManager6.address);
+
+    await tradeExtension.initializeExtension(delegatedManager6.address);
+
+    // SetToken + DelegatedManager: TradeModule pending, TradeExtension initialized
+    setToken7 = await setV2Setup.createSetToken(
+      [setV2Setup.dai.address],
+      [ether(1)],
+      [setV2Setup.issuanceModule.address, tradeModule.address]
+    );
+
+    await setV2Setup.issuanceModule.initialize(setToken7.address, ADDRESS_ZERO);
+
+    delegatedManager7 = await deployer.manager.deployDelegatedManager(
+      setToken7.address,
+      factory.address,
+      methodologist.address,
+      [tradeExtension.address],
+      [operator.address],
+      [setV2Setup.usdc.address, setV2Setup.weth.address],
+      true
+    );
+
+    await setToken7.setManager(delegatedManager6.address);
+
+    await tradeExtension.initializeExtension(delegatedManager7.address);
```

Comment:
instead of doing 7 different flows here should I move this to describe statements below?